Geometry/MTDNumberingBuilder/plugins/CmsMTDLevelBuilder.cc

```diff
@@ -1,31 +1,171 @@
-#include "Geometry/MTDNumberingBuilder/plugins/CmsMTDLevelBuilder.h"
 #include "DetectorDescription/Core/interface/DDFilteredView.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "Geometry/MTDNumberingBuilder/interface/GeometricTimingDet.h"
+#include "Geometry/MTDNumberingBuilder/plugins/CmsMTDLevelBuilder.h"
 #include "Geometry/MTDNumberingBuilder/plugins/ExtractStringFromDDD.h"
-#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
+bool subDetByType(const GeometricTimingDet* a, const GeometricTimingDet* b)
+{
+    // it relies on the fact that the GeometricTimingDet::GDEnumType
+    // enumerators used to identify the subdetectors in the upgrade geometries
+    // are equal to the ones of the present detector + n*100
+    return a->type() < b->type();
+}
+
+// NP** Phase2 BarrelEndcap
+bool phiSortNP(const GeometricTimingDet* a, const GeometricTimingDet* b)
+{
+    if (fabs(a->translation().rho() - b->translation().rho()) < 0.01
+        && (fabs(a->translation().phi() - b->translation().phi()) < 0.01
+               || fabs(a->translation().phi() - b->translation().phi()) > 6.27)
```

Comment:
is 6.27 equal to 2.*M_PI? Please use the latter.

Reply:
Hi, good to learn about M_PI! I'll use it for sure in the other instances, but here it seems to me that the intention was `> M_PI - 0.01`. Still, changing it to that would mean changing a numerical value at the 3rd digit after the leading one, which is maybe not adequate in a technical PR.

Reply:
ok, leave it here as is. I'll make a note to contact the code developer.